crates/hir-ty/src/lib.rs

```diff
@@ -142,6 +144,46 @@ pub type ConstrainedSubst = chalk_ir::ConstrainedSubst<Interner>;
 pub type Guidance = chalk_solve::Guidance<Interner>;
 pub type WhereClause = chalk_ir::WhereClause<Interner>;
 
+/// A constant can have reference to other things. Memory map job is holding
+/// the neccessary bits of memory of the const eval session to keep the constant
+/// meaningful.
+#[derive(Debug, Default, Clone, PartialEq, Eq)]
+pub struct MemoryMap(pub HashMap<usize, Vec<u8>>);
```

Comment:
should that maybe be a BTreeMap for sortedness / iteration order?

Reply:
I don't think this should ever need to be iterated over.

Reply:
Yes it probably doesn't need to be sorted and iterated, the only problem I can think of is `Hash` implementation. I currently excluded memory map field from hash implementation of the `ConstScalar`. If it is fine, then I think it doesn't need to be a `BTreeMap`.